src/main/java/Parser.java

```diff
@@ -0,0 +1,136 @@
+     */
+    public static int markDoneNum(String command) {
+        if (command.length() <= 5) {
+            throw new DukeMissingDescriptionException("Task number missing.");
+        }
+        String num = command.substring(5);
+        try {
+            return Integer.valueOf(num);
+        } catch (NumberFormatException e) {
+            throw new DukeUnknownInputException("Need task NUMBER.");
+        }
+    }
+
+    /**
+     * Gets the description for the To Do task.
+     *
+     * @param command User command.
+     * @return The description of the To Do task.
+     * @throws DukeMissingDescriptionException If there is no description given.
+     */
+    public static String todoDescription(String command) {
+        if (command.length() <= 5) {
+            throw new DukeMissingDescriptionException("Todo description missing.");
+        }
+        return command.substring(5);
+    }
+
+    /**
+     * Gives the description and deadline of the Deadline task.
+     *
+     * @param command User command.
+     * @return A String[] where 0 index contains the description and 1 index contains the deadline.
+     * @throws DukeMissingDescriptionException When no description of deadline is given by user.
+     * @throws DukeUnknownInputException When /by is missing in command.
+     */
+    public static String[] deadlineParams(String command) {
+        if (command.length() <= 9) {
+            throw new DukeMissingDescriptionException("Deadline description and time missing.");
+        }
+        String[] splitted = command.substring(9).split(" /by ", 2);
+        if (splitted.length < 2) {
+            throw new DukeUnknownInputException("Need give/format deadline <description> /by <time>.");
+        }
+        // Check if date is parsable
+        String byWhen;
+        try {
+            LocalDate date = LocalDate.parse(splitted[1]);
+            byWhen = date.format(DateTimeFormatter.ofPattern("MMM d yyyy"));
+        } catch (DateTimeParseException e) {
+            byWhen = splitted[1];
+        }
+        String[] descriptionAndBy = {splitted[0], byWhen};
+        return descriptionAndBy;
+    }
+
+    /**
+     * Gives the description and time of the Event task.
+     *
+     * @param command User command.
+     * @return A String[] where 0 index contains the description and 1 index contains the time of event.
+     * @throws DukeMissingDescriptionException When no description of event is given by user.
+     * @throws DukeUnknownInputException When /at is missing in command.
+     */
+    public static String[] eventParams(String command) {
+        if (command.length() <= 6) {
+            throw new DukeMissingDescriptionException("Event description and time missing.");
+        }
+        String[] splitted = command.substring(6).split(" /at ", 2);
+        if (splitted.length < 2) {
+            throw new DukeUnknownInputException("Need format event <description> /at <time>.");
```

Comment:
Should not create too many user defined exception (new exception classes created by yourself). Only create those that are not provided by Java and really necessary. In this case, it should be DukeException because it correlates with invalid user input for Duke chatbot. In all cases, it will print the error message only, so I don't think you need to create special classes for every single one of them. One DukeException, which takes in the message in the constructor and prints out the error message should be enough.